xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -58,6 +73,12 @@
             nameUri.getAuthority(), "nameUri (%s) doesn't have an authority", nameUri);
   }
 
+  // It's hard to make bootstrapper a final field and pass Bootstrapper.getInstance() to the
+  // constructor argument because Bootstrapper.getInstance() throws.
+  void setBootstrapperForTest(Bootstrapper bootstrapper) {
```

Comment:
Delete this.

```java

@Nullable
private Bootstrapper bootstrap;
XdsNameResolver(String name) {
   this(name, createBootstrapper());
}

@VisibleForTesting
XdsNameResolver(String name, Bootstrapper bootstrapper) {
   ...
}

@Nullable
private static Bootstrapper createBootstrapper() {
   try {
      return Bootstrapper.getInstance();
   } catch (Exception e) {
      logger.log(...);
   }
   return null;
}
```

We are losing exception information when bootstrap fails. But this is ok for now as we have not figure out how the exception should propagate yet.

Reply:
Done.